editor/plugins/animation_player_editor_plugin.cpp

```diff
@@ -1091,9 +1091,27 @@ void AnimationPlayerEditor::_animation_duplicate() {
 		return;
 	}
 
-	String new_name = current;
-	while (player->has_animation(new_name)) {
-		new_name = new_name + " (copy)";
+	int index_digits = 0;
+	for (int i = current.length() - 1; i >= 0; i--) {
+		if (is_digit(current[i])) {
+			index_digits++;
+		} else {
+			break;
+		}
+	}
+	int64_t count = current.right(index_digits).to_int();
+	String new_name = (index_digits) ? current.left(-index_digits) : current;
+	while (true) {
+		String attempt = new_name;
+		if (count > 1) {
+			attempt += itos(count);
+		}
```

Comment:
```suggestion
		if (count > 1) {
			attempt += vformat("_%d", count);
		}
```

Reply:
Same with #48570.

Reply:
The workflow of duplicating animations is quite different.

If we did this way, duplicating a `CoolAnim2` would remove the `2` and append a `_3` result into a `CoolAnim_3`, breaking the style the user has established.

Once again, I'm suggesting to just increase the index if it's already there, and use a `_` *only if the user didn't establish a style*.

Reply:
If you duplicate `CoolAnim2`, it should be `CoolAnim2_2`. If you duplicate `CoolAnim2_2`, it should be `CoolAnim2_3`.